run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
+	seen := map[string]string{}
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// Checks if outputVariables are defined in terms of other outputVariables.
+// Creates a slice of unique model variables that are required to calculate
+// the requested outputVariables.
+func checkForDerivitives(m map[string]string) (map[string]string, []string) {
+	getVariables := []string{}
```

Comment:
Each slice has a length (the number of items in it) and a capacity (the number of items it can grow to without having to move to a new location in memory). In order to initialize an array with a nonzero length or capacity, you need to use the make command. If you want to initialize a slice where both the length and capacity are zero you could use the command ```make([]string,0) or make([]string, 0, 0)```, but the result is the same as doing ```var s []string```, and the convention is to do the latter. I'm not exactly sure why but my guess is that the latter command makes it more clear that there isn't actually any memory being allocated here.

Now that I think about it, probably the best thing to do here is 

``` go 
getVariables := make([]string, 0 , len(m))
```

This is because we know that we will have at least ```len(m)``` output variables, so we can add that many without having to reallocate memory, but if we need to add additional ones the memory will be added automatically by ```append```